sr/comp/scores.py

```diff
@@ -195,9 +196,11 @@ def __init__(
         scores_data: Iterable[ScoreData],
         teams: Iterable[TLA],
         scorer: ScorerType,
+        ranker: RankerType,
         num_teams_per_arena: int,
     ) -> None:
         self._scorer = scorer
+        self._ranker = ranker
```

Comment:
Could/should we instantiate this just once rather than creating a new one each time we want to do the calculation? That gives a natural place in the instances for caching etc.

Reply:
From what I recall the reasoning for the scorer for having separate instances was to discourage caching between matches while making it easy for each instance to store information for a given match. This just follows the same pattern. While at the moment there's only one member on the class, there's a consideration that in future we may add others (hence having a class not just a single function).